crates/rethnet_eth/src/remote.rs

```diff
@@ -97,72 +110,617 @@ impl BlockSpec {
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    GetTransactionByHash(B256),
     #[serde(rename = "eth_getTransactionCount")]
-    TxCount(Address, SerializableBlockSpec),
+    GetTransactionCount(Address, BlockSpec),
+    #[serde(
+        rename = "eth_getTransactionReceipt",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    GetTransactionReceipt(B256),
+    #[serde(rename = "eth_mining")]
+    Mining(),
+    #[serde(rename = "eth_newBlockFilter")]
+    NewBlockFilter(),
+    #[serde(
+        rename = "eth_newFilter",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    NewFilter(FilterOptions),
+    #[serde(rename = "eth_newPendingTransactionFilter")]
+    NewPendingTransactionFilter(),
+    #[serde(rename = "eth_pendingTransactions")]
+    PendingTransactions(),
+    #[serde(
+        rename = "eth_sendRawTransaction",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SendRawTransaction(ZeroXPrefixedBytes),
+    #[serde(
+        rename = "eth_sendTransaction",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SendTransaction(TransactionInput),
+    #[serde(rename = "eth_sign")]
+    Sign(Address, ZeroXPrefixedBytes),
+    #[serde(rename = "eth_signTypedData_v4")]
+    SignTypedDataV4(Address, eth::eip712::Message),
+    #[serde(
+        rename = "eth_subscribe",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    Subscribe(Vec<String>),
+    #[serde(rename = "eth_syncing")]
+    Syncing(),
+    #[serde(
+        rename = "eth_uninstallFilter",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    UninstallFilter(U256),
+    #[serde(
+        rename = "eth_unsubscribe",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    Unsubscribe(Vec<ZeroXPrefixedBytes>),
 }
 
-#[derive(serde::Serialize)]
+#[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
 #[serde(rename_all = "camelCase")]
 struct GetLogsInput {
-    from_block: SerializableBlockSpec,
-    to_block: SerializableBlockSpec,
+    from_block: BlockSpec,
+    to_block: BlockSpec,
     address: Address,
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn help_test_method_invocation_serde(call: MethodInvocation) {
+        let json = serde_json::json!(call).to_string();
+
+        #[derive(Debug, serde::Deserialize)]
+        struct MethodInvocationWithUntypedParams {
+            #[allow(dead_code)]
+            method: String,
+            #[allow(dead_code)]
+            params: Vec<serde_json::Value>,
+        }
+        serde_json::from_str::<MethodInvocationWithUntypedParams>(&json).expect(
```

Comment:
What does this validate?

Reply:
i wrote a comment to explain in [this commit](https://github.com/NomicFoundation/hardhat/pull/3998/commits/4e29b86397af028139e8ce9ed37f2b33a30cff05). let me know what you think.